feat: desk refresh

maharshivpatel opened the pull request:
First draft of frappe framework's new desk design.

> no-docs

maharshivpatel committed:
fix: add Inter V fonts for new design

Added Inter V normal & italic because it used new design instead of Inter. however, Inter is not removed because it may be used in other places.

maharshivpatel committed:
fix: add color variables for espresso

colors, overlays and gradient based on the espresso color palette

maharshivpatel committed:
fix: add drop shadows based for new design

added dropshadows, input focus rings

maharshivpatel committed:
fix: add files spacing and borders as per new design

added Spacing and borders files for espresso design

maharshivpatel committed:
fix: add font styles for new design

extracted and added text styles from espresso design.

maharshivpatel committed:
fix: add espresso icons

added espresso icons svg sprite and loaded them in base and app

maharshivpatel committed:
fix(minor): updated variables file to use espresso files

Imported espresso files in variables file

maharshivpatel committed:
fix: update website / login page design

changed login page style and updated variables for website pages

maharshivpatel committed:
fix(minor): change icons to espresso icons

changed email and password icons to espresso icons

maharshivpatel committed:
fix: update navbar size and styles

changed navbar size, icons, and styles.

maharshivpatel committed:
fix: add espresso icons in frappe.utils.icon

this will need to clean this at the end.

maharshivpatel committed:
fix: update css_variable file for new design

removed redundant variables from the file and changed few to match the new design

maharshivpatel committed:
fix: update avatar style and notification popover

updated size and style of avatar and notification popover

maharshivpatel committed:
fix: update old icons to match new primary color

some of the old icons have blue as primary in them updated to match the new primary color

maharshivpatel committed:
fix: update page styles to new design

- add page head styles and changes icons to espresso

maharshivpatel committed:
fix: update indicator styles to match new design

changed variables and styles to match new design

maharshivpatel committed:
fix: updated workspace desktop and global css

changed few global styles and updated workspace styles according to new design

maharshivpatel committed:
fix: updated onboarding widget style

removed drop shadow and updated styles to match the new design

maharshivpatel committed:
fix(minor): updated workspace widgets

- shortcut widget changed indicator style
- quick list widget updated icon
- onboarding widget changed step no to tick icon
- number card changed percentage text structure

maharshivpatel committed:
fix: update workspace icons for new designs

maharshivpatel committed:
fix: workspace edit mode and private sidebar indicator

maharshivpatel committed:
fix(minor): import _borders file in variables

maharshivpatel committed:
fix: update buttons, breadcrumb and global scss

- updated some styles in global scss
- updated buttons scss
- updated breadcrumb style to be consistent with new design

maharshivpatel committed:
fix(minor): button text style

maharshivpatel committed:
fix: update form, controls, dropdown styles

- update form text style
- navbar menu padding
- added switch style to checkbox TODO: add option to configure switch.
- misc fixes

maharshivpatel committed:
fix: update icons to espresso for different pages

updated icons on various parts of the page to espresso

maharshivpatel committed:
fix(minor): update list view to new design

- Looks better but still needs some work

maharshivpatel committed:
fix: update form, grid and sidebar to new design

updated design for form, grid and sidebar.
- sidebar only empty state is done need to work on other states.
- grid height feels too much, need to reduce it.

maharshivpatel committed:
fix: update quill editor

updated quill editor styles to match new design

maharshivpatel committed:
fix(minor): updated minimal styles in datatable for new design

for now i have made minimum changes need to change height of datatable
datatable rows are absolute positioned so need to change that.

maharshivpatel committed:
fix: update form sidebar html for new design

maharshivpatel committed:
fix: updated comment and activity section

maharshivpatel committed:
Merge branch 'develop' into feat-desk-refresh

maharshivpatel commented:
- [x] setup wizard 😅
- [x] Dark Mode
- [x] user-profile page
- [x] form page sidebar handle all states
- [x] User Avatar component
- [x] @shariquerik can you make private workspace indicator color configurable.
- [x] datatable inline filter input caret is hidden due to border radius

ankush commented:
Surface level review based on 10-20 minutes of use:
- [x] Workspace spacing doesn't feel consistent
- [x] Grid spacing is too much IMO
- [x] Sidebar files:
    - [x] Attach file showing two versions of adding file
    - [x] Folder icon is missing, check this: https://github.com/frappe/frappe/pull/21819
- [x] User avatars everywhere are not showing the circular frame when user doesn't have profile pic.
- [x] Onboarding skipped rows show blue tick (should be gray/black)
- [x] Sidebar tags spacing is not consistent almost touching some other element above it
- [x] list view reload button title is "es line reload" should be just reload

shariquerik commented:
<img width="1280" alt="image" src="https://github.com/frappe/frappe/assets/30859809/2aeea327-0e0f-4b4a-b4e5-86ad2c289a84">

1. The timeline line looks out of place
2. The Avatar without an image looks odd maybe keep the border
3. The cursor should start from "Type..." placeholder

<img width="236" alt="image" src="https://github.com/frappe/frappe/assets/30859809/f792c70b-f3ba-4e3f-8131-f1d4c5516466">

Also, keep this circle from the old design before the New Email button

ruchamahabal commented:
Great work 👏 

Few things I noticed:

### Workspace

1. Workspace > Shortcuts > Title and Count badge alignment seems off. The count indicator seems to be placed a bit lower than the link. Can we have badges here, like the current view?
![image](https://github.com/frappe/frappe/assets/24353136/fb8c9b8a-3d78-4c97-8af8-6548e64ab14f)
![image](https://github.com/frappe/frappe/assets/24353136/9100af41-f357-4eb0-99d8-2adffcfbb4ee)

### Forms

2. Font size for section descriptions (field level descriptions are working fine) 
![image](https://github.com/frappe/frappe/assets/24353136/dbc4cb47-ab3a-413e-b7f8-ea66acf14c59)

### Child Table

3. Child table cells look huge compared to other form fields— also, the edit button & icon alignment, and padding need fixing IG.
![image](https://github.com/frappe/frappe/assets/24353136/9c319b40-a91b-484e-ab92-1c6b76a9da4d)

4. Also the link indicators inside child table link fields:
![image](https://github.com/frappe/frappe/assets/24353136/599c1559-4048-43f8-a313-d98a04cc83af)

5. Configure Columns view:
![configure-cols](https://github.com/frappe/frappe/assets/24353136/7e7ce63b-e270-4df0-9a49-9bd628cbee4e)

### Notifications

6. Shows 2 scrollbars. The header should be sticky.
![image](https://github.com/frappe/frappe/assets/24353136/9453432f-ed96-4198-a9be-e4cf1c07f3f1)

### Dashboard

7. Dashboard items should be cards IMO. Introduces some delay in figuring out which element belongs to which item

![image](https://github.com/frappe/frappe/assets/24353136/053f85c0-2f3a-4437-9388-71f371b33db6)

deepeshgarg007 commented:
<img width="720" alt="image" src="https://github.com/frappe/frappe/assets/42651287/da9db9eb-db46-46c0-8ffb-3cdefcb27522">

<img width="485" alt="image" src="https://github.com/frappe/frappe/assets/42651287/bdf0cf1e-1bd2-4e30-baa3-bf47797b9f82">

<img width="711" alt="image" src="https://github.com/frappe/frappe/assets/42651287/21f76971-ccf6-4aee-950e-06b472f356ae">

<img width="681" alt="image" src="https://github.com/frappe/frappe/assets/42651287/628764e7-e27d-4069-b520-e3122ba66f7e">

shariquerik committed:
fix: made workspace indicator color customizable

maharshivpatel committed:
fix: form comment and timeline fixes

- Activty heading was added under has_communications condition which broke layout when there were no communications
- comment have user content so background color to white as suggested by @netchampfaris
- fixed postion of Activty and icons / dots in timeline

maharshivpatel committed:
fix(minor): remove unused timeline css

maharshivpatel committed:
fix: issues shared by frappe team's initial review

fixed issues shared in comment
- Refresh Icon Title ( also added safe guard to strip es prefix )
- changed card shadow variable to more subtle shadow
- reduced padding for padding-sm variable
- removed fixed height which caused overflow of long content for class like-disabled-input
- changed frappe-card box-shadow to border to have more flat look
- added green color back to avatar
- changed form-section-description text size as per new text-size
- fixed double scrollbar issue in notification list

pateljannat commented:
Website Theme form is breaking ( disabled large input ).

<img width="1440" alt="Screenshot 2023-08-28 at 12 05 41 PM" src="https://github.com/frappe/frappe/assets/31363128/601e6559-3665-4616-a019-fac8071c0246">

maharshivpatel committed:
fix: handle onboarding widget different states

- replaced icons to new espresso icons and changed css to match new design.
- modified icon util to add es-solid css class if icon is from es-solid set.

maharshivpatel committed:
fix(minor): minor style changes

- changed file uploaded icon colors
- added outline for autofill state of login

maharshivpatel committed:
fix: workspace widgets for new design

- updated shortcut link icon and indicator location as per new design
- changed text selection color to match the overall black as primary theme
- added border to dashboard-widget-box to have structure
- updated links widget box as new design was looking too empty

maharshivpatel committed:
fix(minor): update style for Configure Columns modal

maharshivpatel committed:
fix: grid padding and icon positions

- made grid more compact
- changed top postion of icon to match the new size
- ace_editor input have default height of 48 so added that as min height
- changed select field padding top and bottom to 0

maharshivpatel committed:
fix: list view hover state and icons
- updated icons and style for hover and focus state.

maharshivpatel committed:
fix: change datatable styles to match new design

maharshivpatel committed:
fix: updated sidebar

- fixed attachment button
- made button text ellipsis
- fixed margin and padding for shared and assigned avatars

ankush commented:
Contrast for field labels is quite low 

![image](https://github.com/frappe/frappe/assets/9079960/c3f0d2df-07f0-47b6-b6b8-94802d774487)



![image](https://github.com/frappe/frappe/assets/9079960/220313c1-ed29-49f8-b5a1-a661e0bad018)

maharshivpatel committed:
fix: contrast issue on major/prominent parts of design

changed light text to muted from gray-600  to gray-700.
misc: fixed grid-heading alignment issue caused by line-height

shariquerik committed:
style: cosmetic changes in form builder

shariquerik committed:
style: changed primary arrow color

maharshivpatel committed:
fix: typography added letter-spacing

* added function to calculate letterspacing based on espresso
* added mixin to add font-size, font-weight and letter-spacing
* letter spacing is based on size and weight
* changed letter spacing on major places
* misc: minor kanban style fixes and changed border radius for .page-card

maharshivpatel committed:
Merge branch 'develop' into feat-desk-refresh

maharshivpatel committed:
fix: workspace title and onboarding widget typography

* After changing text styles to get_textstyle mixin needed to change few text sizes.

barredterra commented:
Can we remove these meaningless leading icons (or replace them with meaningful ones)?

With:

![before](https://github.com/frappe/frappe/assets/14891507/770a6770-5643-4004-a4f6-61e18338d5c4)

Without:

![after](https://github.com/frappe/frappe/assets/14891507/7fc587f8-165b-4ad4-b967-82178d9aa699)

PR: https://github.com/maharshivpatel/frappe/pull/8

pateljannat commented:
MultiSelect Field values are not clear.

<img width="512" alt="Screenshot 2023-09-11 at 12 07 30 PM" src="https://github.com/frappe/frappe/assets/31363128/0d9237e3-f700-468b-b3f6-89da9c260f98">

maharshivpatel committed:
fix: dark mode, timeline, logo colors

* changed variables from gray-* to dark mode compatible ones.
* timeline style fixes
* change logo with new black colors.

maharshivpatel committed:
Merge branch 'develop' into feat-desk-refresh

ankush commented:
Dark mode:

Read only and editable checkboxes look same. 
![image](https://github.com/frappe/frappe/assets/9079960/ac5dbf2e-1221-4c9a-a4f0-0ee4e33d2b38)


Follow link button has weird background colour here
![image](https://github.com/frappe/frappe/assets/9079960/5e4198bf-03a4-48d2-9b53-536b95408357)

maharshivpatel committed:
fix: dark mode and misc

* timeline, quill toolbar, attach control, modal backdrop.
* datatable, notification color and spacing, checkbox disabled state and more.
* workspace edit mode, grid link icon, label and input spacing.

ankush committed:
Merge branch 'develop' into feat-desk-refresh

ankush committed:
test: update tests for ui changes

- update selectors
- remove color check code

ankush commented:
Datatable dropdown has weird hover background. S

![image](https://github.com/frappe/frappe/assets/9079960/257c9a0d-8679-4bcf-85a8-3de23809e84f)

barredterra commented:
![Bildschirmfoto 2023-09-18 um 18 06 52](https://github.com/frappe/frappe/assets/14891507/998222d2-5823-421f-895c-e35563c8f4f8)

- [ ]  The "+" is blurry
- [x]  The "List View" icon doesn't look vertically centered
- [ ]  The "Menu" tooltip is not horizontally centered
- [ ]  The menu overlay has a visible `dropdown-divider` on top, despite being unused

---

![Bildschirmfoto 2023-09-18 um 18 12 35](https://github.com/frappe/frappe/assets/14891507/5f262414-8e70-4ecc-9edd-ff0ebd278f0c)

- [x] The "+ Add Asset" button becomes white on click (`:active:hover:focus`)

---

![Bildschirmfoto 2023-09-18 um 18 37 19](https://github.com/frappe/frappe/assets/14891507/c16e3838-f5d0-4657-a18c-3aff1e77af67)

- [x] This looks slightly crammed and confusing because it doesn't vertically line up with anything. And we have a random dot in the middle. Why not keep the old position or align it with the comment field? For example:

![Bildschirmfoto 2023-09-18 um 18 28 01](https://github.com/frappe/frappe/assets/14891507/516a8941-1284-441a-b87a-3b2cab21e010)

---

![Bildschirmfoto 2023-09-18 um 18 30 47](https://github.com/frappe/frappe/assets/14891507/983665bf-3d15-43ce-94ff-9684c952dc1e)

- [x] The comment button is not aligned with comment field. 
Maybe we could remove it entirely and submit comments by hitting enter?

---


The table header in the list view header looks much better in **bold**, IMHO. My desired look:

![Bildschirmfoto 2023-09-18 um 18 46 17](https://github.com/frappe/frappe/assets/14891507/eefdbe85-66c4-4340-bed0-48868e89c842)

maharshivpatel committed:
fix: don't override label in set_action when icon is present

maharshivpatel committed:
fix: cypress, dark mode, grid search, timeline

fixes for Espresso
* due to have fixed height of the grid heading, the search bar was not visible now
  added class that will double the height of the grid heading row if filter row is present.
* dark mode fixes: button active state, Awesomeplete dropdown, Quill editor.
* timeline style fixes

maharshivpatel committed:
fix: awesome_bar & sidebar cypress test as per espresso.

maharshivpatel committed:
Merge branch 'develop' into feat-desk-refresh

maharshivpatel committed:
fix(minor): dark mode checkbox disabled state.

maharshivpatel committed:
fix(minor): form tour driver dark theme background color.

maharshivpatel committed:
fix: datepicker style as per espresso

surajshetty3416 commented:
@mergifyio update

maharshivpatel committed:
fix: sidebar tags as per espresso

changed frappe.ui.Tags to sidebar button to match espresso style.

maharshivpatel committed:
fix(minor):  datatable header cell bg color

git-avc commented:
Amazing new look! Congrats!
IMHO, maybe section title font size should be little smaller . Right now is bigger than tab title or other elements ...

![image](https://github.com/frappe/frappe/assets/94137451/b21b0354-5fb6-409a-b947-765d12220461)

maharshivpatel committed:
chore: fix linters

maharshivpatel commented:
> Amazing new look! Congrats! IMHO, maybe section title font size should be little smaller . Right now is bigger than tab title or other elements ...
> 
> ![image](https://user-images.githubusercontent.com/94137451/269548563-b21b0354-5fb6-409a-b947-765d12220461.png)

I do agree with you but making it small will make input text bigger than section heading which is worse UX IMO.

maharshivpatel committed:
chore: remove form builder not important and flaky ui test.

maharshivpatel committed:
fix(minor): change comment box color to control bg

vorasmit commented:
I appreciate all the work done with UI but to be honest, there was no WOW factor involved with the UI revamp. Moreover, it felt rough and unpleasing to the eyes in the first go. Also, it was difficult for me to understand the objective or motive for the revamp and what problems the new UI was solving.

## General Comments
 
![image](https://github.com/frappe/frappe/assets/10496564/be8c5d0a-6c12-416b-99d9-cc57d2b64737)

I am no UI expert to comment on UI elements and design sense used, but I will add a few comments based on what I felt.
- Colours used in old UI were more pleasing to the eyes. Brighter white and black are not pleasing. Also, the colours used for status were a little brighter.
- Increased border-radius with reduced padding or margin looks unprofessional IMO (especially for fields and buttons).
- Too many variations of Font size and font weight were used.
- Alignment for buttons doesn't look appropriate.
- Background (separating form or elements) was useful. It helped narrow the vision on the main elements.

Separate Background is Useful
![image](https://github.com/frappe/frappe/assets/10496564/b8d62a22-d2f9-4ff3-adf8-7d5980d9d4cd)

![image](https://github.com/frappe/frappe/assets/10496564/8ae8f1ba-78f6-4fa3-bd63-abf60ec4f2fa)

## Specific Comments
### Less Intuitive Icons

- The icon used before for "assigned to" looks more appropriate than the one used now
- The icon for "shared with" along with the word used `Invite` don't seem to go with each other

![image](https://github.com/frappe/frappe/assets/10496564/ce16ccb9-0394-47a0-9265-57d993d7f055)  ![image](https://github.com/frappe/frappe/assets/10496564/d6b4d409-8a83-42ce-a158-3b9ccbd32df3)

### List view

- Earlier 5 filters were used to fit. Now there is empty space always left. By default most forms had 5 filters and unnecessary extra column for all filters doesn't look good.
- The design for the background of the list title doesn't fit here. Maybe because of a line divider just above. It could touch all borders if required.

![image](https://github.com/frappe/frappe/assets/10496564/e796b899-ff35-42f8-98f8-eab0e55bdc9c)

### Elements not falling in place.

Number Cards
![image](https://github.com/frappe/frappe/assets/10496564/1bdbf73f-e66d-4c79-ade0-85c5f59a7920)

Table Filters
![image](https://github.com/frappe/frappe/assets/10496564/a2d37f7a-d847-40ab-8e04-cb744895f051)

Headers for Datatable
![image](https://github.com/frappe/frappe/assets/10496564/a87d8338-e444-409e-8a91-c5d17724c2da)

These are a quick few things I felt were going wrong based on a quick review of the new UI.

sintayehu156 commented:
i think the previous UI was Very attractive than the new one, the padding, the icons, the button colors, are not good.

ankush commented:
post feedback/issue here: https://github.com/frappe/frappe/issues/22508
